<?xml version="1.0"?>
<launch>
	<!-- Which Algorithm -->
	<param name="tracking_method" value="SORT"/>
	
	<!-- Algorithm parameters for SORT  -->
	<param name="max_age" value="3"/>  <!-- Max age without new predictions before killing a tracker -->
	<param name="min_hits" value="3"/> <!-- Minimum number of hits before a tracker is used as a prediction -->
	<param name="min_iou_thresh" value="4"/>
	
	<!-- Number of classes, a SORT tracker is created for each class to prevent overlapping boxes from different classes from interacting -->
	<param name="num_classes" value="80"/>  
	
	<!-- Topic name parameters -->
	<param name="object_detection_topic" value="/perception/lidar/bboxes3d"/>
	<param name="tracked_objects_topic" value="/perception/lidar/tracked_bboxes3d"/>
	
	<!-- Logging parameters    -->
	<param name="log_info" value="False"/> <!-- Log the inference time --> 
	
	<!-- Plotting node (Publishes marker message to visualize bounding boxes in ROS -->
	<node name="tracking_wrapper" pkg="sort_3d" type="tracking_3d_wrapper.py" output="screen"/>
	
	<!-- Plotting Node -->
	<include file="$(find sort_3d)/launch/plotter.launch"/>
	
	
	<!-- TODO (Minor Improvements): 
	1) Use rospy time difference in the tracker instead of assuming constant time differences
	-->
</launch>
